### PR TITLE
Address review comments for deprecated warning on NSObject.hashValue override

### PR DESCRIPTION
**Pull request by DougGregor**
Address the review comments from @jrose-apple and @lorentey on https://github.com/apple/swift/pull/18407, cleaning up my newly-introduced warning for overriding `NSObject.hashValue`

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: lib/Sema/TypeCheckDeclOverride.cpp [hunk 1]
```diff
@@ -1524,8 +1524,10 @@ static bool checkSingleOverride(ValueDecl *override, ValueDecl *base) {
   if (auto baseVar = dyn_cast<VarDecl>(base)) {
     if (auto classDecl =
           baseVar->getDeclContext()->getAsClassOrClassExtensionContext()) {
-      if (classDecl->getBaseName().userFacingName() == "NSObject" &&
-          baseVar->getBaseName().userFacingName() == "hashValue") {
+      if (baseVar->getName() == ctx.Id_hashValue &&
+          classDecl->getName().is("NSObject") &&
+          (classDecl->getModuleContext()->getName().is("ObjectiveC") ||
+           classDecl->getModuleContext()->getName().is("Foundation"))) {
```
Comment (jrose-apple) [inline lib/Sema/TypeCheckDeclOverride.cpp]: I believe these two are also KnownIdentifiers. (We're not hugely consistent about what we use those for. Maybe it's less important for checking anyway.)

Comment (DougGregor) [reply lib/Sema/TypeCheckDeclOverride.cpp]: So they are!

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Pull request merged**